xwiki-commons-core/xwiki-commons-jakartabridge/xwiki-commons-jakartabridge-servlet/src/main/java/org/xwiki/jakartabridge/servlet/JakartaServletBridge.java

```diff
@@ -0,0 +1,1002 @@
+        if (javax == null) {
+            return null;
+        }
+
+        return new MultipartConfigElement(javax.getLocation(), javax.getMaxFileSize(), javax.getMaxRequestSize(),
+            javax.getFileSizeThreshold());
+    }
+
+    //////////////////////////////////////////////////
+    // Multi
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Collection<javax.servlet.http.Part> toJavax(Collection<Part> jakarta)
+    {
+        return JakartaBridge.toJavax(jakarta, JavaxToJakartaPart::new, JakartaToJavaxPart::new);
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Collection<Part> toJakarta(Collection<javax.servlet.http.Part> jakarta)
+    {
+        return JakartaBridge.toJakarta(jakarta, JavaxToJakartaPart::new, JakartaToJavaxPart::new);
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Enumeration<javax.servlet.Servlet> toJavax(Enumeration<Servlet> jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        return Collections.enumeration(Collections.list(jakarta).stream().map(JakartaServletBridge::toJavax).toList());
+    }
+
+    /**
+     * @param javax the javax version
+     * @return the javax version
+     */
+    public static Enumeration<Servlet> toJakarta(Enumeration<javax.servlet.Servlet> javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        return Collections.enumeration(Collections.list(javax).stream().map(JakartaServletBridge::toJakarta).toList());
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static javax.servlet.http.Cookie[] toJavax(Cookie[] jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        javax.servlet.http.Cookie[] javax = new javax.servlet.http.Cookie[jakarta.length];
+
+        for (int i = 0; i < jakarta.length; ++i) {
+            javax[i] = toJavax(jakarta[i]);
+        }
+
+        return javax;
+    }
+
+    /**
+     * @param javax the javax version
+     * @return the javax version
+     */
+    public static Cookie[] toJakarta(javax.servlet.http.Cookie[] javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        Cookie[] jakarta = new Cookie[javax.length];
+
+        for (int i = 0; i < javax.length; ++i) {
+            jakarta[i] = toJakarta(javax[i]);
+        }
+
+        return jakarta;
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static EnumSet<javax.servlet.DispatcherType> toJavax(EnumSet<DispatcherType> jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        return EnumSet.copyOf(jakarta.stream().map(JakartaServletBridge::toJavax).collect(Collectors.toList()));
```

Comment:
Why don't you directly collect the result of the stream in an `EnumSet`?

```suggestion
        return jakarta.stream().map(JakartaServletBridge::toJavax).collect(Collectors.toCollection(() -> EnumSet.noneOf(javax.servlet.DispatcherType.class)));
```

Reply:
I just did not notice this way to use collect yet.